### PR TITLE
Fix: Support for dash and slash in `valid-jsdoc` (fixes #1598)

### LINKED ISSUE #1598: valid-jsdoc rule is too strict
here is a simple class jsdoc comments:

```js
/**
 * Base component implementation.
 *
 * @constructor
 * @extends module:stb/emitter~Emitter
 */
```

it is fully recognizable by jsdoc generator but valid-jsdoc rule is telling it's wrong

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/7189848-valid-jsdoc-rule-is-too-strict?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by gyandeeps**

### CODE REVIEW

File: tests/lib/rules/valid-jsdoc.js [hunk 1]
```diff
@@ -339,6 +339,35 @@ ruleTester.run("valid-jsdoc", rule, {
...
+        {
+            code:
+            "/**\n" +
+            " * Test dash and slash.\n" +
+            " * @extends module:stb/emitter~Emitter\n" +
+            " */\n" +
+            "function foo() {}",
+            options: [{
+                "prefer": {
+                    "returns": "return"
```
Comment (platinumazure) [inline tests/lib/rules/valid-jsdoc.js]: Is the `prefer` config necessary for this test?

File: tests/lib/rules/valid-jsdoc.js [hunk 2]
```diff
@@ -339,6 +339,35 @@ ruleTester.run("valid-jsdoc", rule, {
...
+            code:
+            "/**\n" +
+            " * Test dash and slash.\n" +
+            " * @requires module:config\n" +
+            " * @requires module:modules/notifications\n" +
+            " */\n" +
+            "function foo() {}",
+            options: [{
+                "prefer": {
+                    "returns": "return"
```
Comment (platinumazure) [inline tests/lib/rules/valid-jsdoc.js]: Is the `prefer` config necessary for this test?

File: tests/lib/rules/valid-jsdoc.js (hunk 1 above)
Comment (gyandeeps) [inline tests/lib/rules/valid-jsdoc.js]: nope. Just copy paste carryover from an existing test. Doesnt hurt tho.

Comment (platinumazure) [inline tests/lib/rules/valid-jsdoc.js]: I know it doesn't hurt, but readers not in the know (such as me) might assume it's relevant to the test. This option imposes am unnecessary cognitive burden. Just my opinion but I think it should go.

Comment (nzakas) [inline tests/lib/rules/valid-jsdoc.js]: Agree with @platinumazure , best to keep test conditions narrow.

**Comment by gyandeeps** [comment]
Fixed. :+1:

**Comment by nzakas** [comment]
Lgtm

**Pull request merged**